purchases/src/main/kotlin/com/revenuecat/purchases/models/StoreProduct.kt

```diff
@@ -88,4 +90,16 @@ interface StoreProduct {
      * Creates a copy of this `StoreProduct` with the specified `offeringId` set.
      */
     fun copyWithOfferingId(offeringId: String): StoreProduct
+
+    /**
+     * Null for INAPP products. The price of the [StoreProduct] in the given locale in a monthly recurrence.
+     * This means that, for example, if the period is annual, the price will be divided by 12.
+     * It uses a currency formatter to format the price in the given locale.
+     * Note that this value may be an approximation.
+     * For Google subscriptions, this value will use the basePlan to calculate the value.
+     * @param locale Locale to use for formatting the price. Default is the system default locale.
+     */
+    fun formattedPricePerMonth(locale: Locale = Locale.getDefault()): String? {
+        return period?.let { price.formattedPricePerMonth(it, locale) }
```

Comment:
will the Java version fail compiling? or just not have the default implementation? Java 8 has default implementations for interfaces though right?

Reply:
It will compile, the only difference is that, if a developer tries to create an implementation of `StoreProduct` in Java, they would need to implement this method, but not in kotlin.

As for Java 8 supporting default implementations, yes, that's true but the way kotlin does default implementations is different. Small article as reference: https://proandroiddev.com/kotlin-interface-default-implementation-how-does-it-work-3af5056e0c03

Reply:
Oh interesting